### PR TITLE
Bidirectional GitHub-Agamemnon sync with webhooks and reconciliation

### LINKED ISSUE #165: No bidirectional sync: GitHub-side edits not reflected in-store
The current implementation delivers write-through (Agamemnon → GitHub) and one-time rehydration at startup. Changes made directly to GitHub Issues (e.g. manual status label edits, external tooling) are never picked up after the initial hydration. A webhook receiver on POST /v1/github/webhook (or a periodic poll) is needed to close this gap. This was explicitly deferred in the §3 plan but should be tracked as a follow-on issue.

_Follow-up from #44_

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary

Implement bidirectional synchronization between GitHub Issues and the Agamemnon in-memory Store via:
- HMAC-SHA256 webhook verification for real-time event processing
- Periodic reconciliation polling as a safety net for missed webhooks  
- Last-writer-wins conflict resolution via RFC 3339 timestamp comparison
- Soft-delete semantics for closed issues (preserves recovery)

Closes #165

### New Files

1. **include/projectagamemnon/github_webhook.hpp** / **src/github_webhook.cpp**
   - `verify_github_signature()`: Constant-time HMAC-SHA256 verification
   - `normalize_issues_event()`: Filters and normalizes GitHub issues events
   - Supports: opened, edited, closed, reopened, labeled, unlabeled actions

2. **test/src/test_github_webhook.cpp**
   - 12 tests covering HMAC verification, payload normalization, and edge cases

### Modified Files

- **src/store.cpp** / **include/projectagamemnon/store.hpp**
  - `apply_github_event()`: Single ingestion point for webhook and reconciler
  - `reconcile_from_github()`: Periodic polling from GitHub API
  - `pick_map_()`: Typed helper for entity-label-to-map dispatch
  - `updatedAt` stamping on creation and update paths
  - Protected `updatedAt` from caller overwrite (skip-set in update methods)
  - Last-writer-wins via RFC 3339 timestamp comparison
  - Soft-delete for closed actions (status="closed", no erase)

- **src/metrics.cpp** / **include/projectagamemnon/metrics.hpp**
  - `record_inbound_sync()`: Observability counter for sync events
  - Outcome labels: applied, skipped_stale, closed, reopened

- **src/routes.cpp**
  - Pre-routing exemption for `/v1/github/webhook` (bypasses API-key auth)
  - `POST /v1/github/webhook` handler with HMAC verification
  - Fail-secure: 503 if `GITHUB_WEBHOOK_SECRET` not configured
  - Returns `{"applied": true/false, "action": "..."}` on success

- **src/server_main.cpp**
  - `std::jthread reconciler` in main() with stop_token
  - Signal handler integration via `g_reconciler->request_stop()`
  - Configuration: `GITHUB_RECONCILE_INTERVAL_SEC` (default 300s)
  - Graceful shutdown via jthread automatic join

- **CMakeLists.txt** / **test/CMakeLists.txt**
  - Added `github_webhook.cpp` to core target
  - Added `test_github_webhook.cpp` to test suite

### Environment Variables

- `GITHUB_WEBHOOK_SECRET`: Secret key for HMAC signature verification (REQUIRED if webhook enabled)
  - Handler returns 503 if not set
  - Loaded once at startup
- `GITHUB_RECONCILE_INTERVAL_SEC`: Reconciliation polling interval in seconds (default: 300)
  - Set to 0 to disable polling
  - Only runs if `GITHUB_TOKEN` is configured

### GitHub Webhook Setup

1. Navigate to repository Settings → Webhooks
2. Add webhook:
   - Payload URL: `https://<agamemnon>/v1/github/webhook`
   - Content type: `application/json`
   - Events: Issues only
   - Secret: Set to the value of `GITHUB_WEBHOOK_SECRET`

### Conflict Resolution (Last-Writer-Wins via Timestamp)

For `edited`, `opened`, `labeled`, `unlabeled` actions:
1. Compare GitHub's `updated_at` (RFC 3339) with local `updatedAt`
2. Apply only if GitHub timestamp is newer (lexicographic comparison)
3. If local has no timestamp or empty, apply (cold-start case)
4. Metrics record outcome: `applied` or `skipped_stale`

### Closed/Reopened Actions (Terminal State)

- `closed`: Sets `status="closed"`, `closedAt=updated_at` (no erase, recoverable)
- `reopened`: Restores `status="active"`, re-inserts if missing (idempotent)
- Always applied regardless of timestamps (terminal transitions must win)

### Reconciliation Safety Net

- Runs every `GITHUB_RECONCILE_INTERVAL_SEC` seconds
- Calls `list_issues(label)` for each entity type
- Applies as `edited` action (picks up state changes, label edits, body edits)
- Logs: `[agamemnon] reconcile applied N GitHub edits`
- Errors logged but don't crash the server

## Test Coverage

- 12 webhook unit tests (HMAC, normalization, edge cases)
- Existing Store tests still pass (394/396 passing; 2 pre-existing failures unrelated)
- All 12 GitHub webhook tests passing
- Integration via existing routes/store test patterns

## Security

- Constant-time HMAC comparison via `CRYPTO_memcmp`
- Fail-secure: Webhook disabled (503 response) if secret not configured
- Signal handler guards against use-after-free (pointers nulled post-listen)
- Pre-routing exemption prevents API-key middleware from rejecting webhook

## Rollback

To fall back to startup-only hydration:
1. Unset `GITHUB_WEBHOOK_SECRET` (webhook returns 503)
2. Set `GITHUB_RECONCILE_INTERVAL_SEC=0` (disables polling)
3. No code changes required; existing behavior preserved

### CODE REVIEW

File: src/server_main.cpp [hunk 1]
```diff
@@ -151,6 +176,7 @@
   std::atomic<bool> shutdown_requested{false};
   g_shutdown_flag = &shutdown_requested;
   g_server = &server;
+  g_reconciler = &reconciler;
```
[the remark this answers is not in the document]
Comment (mvillmow) [reply src/server_main.cpp:179]: CodeQL false-positive. `shutdown_requested` is a `std::atomic<bool>` local to `main()` whose lifetime spans the entire program — `main()` does not return until after shutdown completes. The signal handler only dereferences `g_shutdown_flag` while the program is live, and the pointer is explicitly set back to `nullptr` before `main()` returns (see the cleanup that nulls the static pointers so late signals are no-ops). There is no access to the local after its scope ends, so the stored address can never dangle. Resolving as a known false-positive for the signal-handler global-pointer pattern.
